pkg/p2p/libp2p/libp2p.go

```diff
@@ -328,14 +338,18 @@ func (s *Service) Connect(ctx context.Context, addr ma.Multiaddr) (overlay swarm
 		return swarm.Address{}, fmt.Errorf("handshake: %w", err)
 	}
 
-	if err := helpers.FullClose(stream); err != nil {
-		return swarm.Address{}, err
-	}
-
 	if exists := s.peers.addIfNotExists(stream.Conn(), i.Address); exists {
+		if err := helpers.FullClose(stream); err != nil {
+			return swarm.Address{}, err
+		}
+
 		return i.Address, nil
 	}
 
+	if err := helpers.FullClose(stream); err != nil {
+		return swarm.Address{}, err
```

Comment:
`return nil, err`, not a zero address and error

Reply:
I don't think it is possible as `Connect` function returns `swarm.Address`?

Reply:
true. i didn't realise we switched to address which stores the byte slice in a struct (in the old codebase it was just a type alias to a byte slice). resolving :+1: